src/frag.js

```diff
@@ -176,6 +176,14 @@ function insertBefore(
 	insertNode,
 	insertBeforeNode,
 ) {
+	if (
+		insertNode.parentElement
+		&& insertNode.nodeType === Node.ELEMENT_NODE
+		&& $fakeParent in insertNode
+	) {
+		return insertNode;
+	}
```

Comment:
Tests seem to pass without these. When was this necessary?

Reply:
Seems like the second expression could be simplified too, I've pushed an update

Reply:
Looks like tests still pass even with removing the entire if-block. When is it necessary?

Reply:
Try to add frag child node after keep-alive and `insertBefore` will be called instead of `appendChild`
```vue
<app>
  <frag v-frag>
    <keep-alive>
      <component :is="component"></component>
    </keep-alive>1
  </frag>
</app>
```

Reply:
Gotcha, can you add it as a test?

Reply:
Done